akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/Behaviors.scala

```diff
@@ -326,13 +326,12 @@ object Behaviors {
    * @param behavior The actual behavior handling the messages, the MDC is used for the log entries logged through
    *                 `ActorContext.log`
    *
-   * See also [[akka.actor.typed.Logger.withMdc]]
    */
   def withMdc[T](
       interceptMessageClass: Class[T],
-      mdcForMessage: akka.japi.function.Function[T, java.util.Map[String, Any]],
+      mdcForMessage: akka.japi.function.Function[T, java.util.Map[String, String]],
```

Comment:
How about we keep this a `Map[String, Object]` and make String of the objects in the impl?

Reply:
Wouldn't that be surprising for users that we are silently discarding informat (and relying on `toString`).
If it's limited to strings I think we should make that explicit all the way.

Reply:
Yes, you are right, I change my mind!

Reply:
I'll leave to `Map[String, String]` then